compiler/rustc_infer/src/infer/error_reporting/suggest.rs

```diff
@@ -461,6 +461,62 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
             }
         }
     }
+
+    /// For "one type is more general than the other" errors on closures, suggest changing the lifetime
+    /// of the parameters to accept all lifetimes.
+    pub(super) fn suggest_for_all_lifetime_closure(
+        &self,
+        span: Span,
```

Comment:
We probably need to pass the closure's HIR representation in as an argument instead of just using `def_span`, in order to make sure we have the right span (just the stuff in the args `|...|`) and to make sure we don't overwrite the arguments during suggestion.